A3A/addons/scrt/Encounter/fn_encounter_HeliSlingloadCargo.sqf

```diff
@@ -0,0 +1,256 @@
+#include "..\defines.inc"
+FIX_LINE_NUMBERS()
+
+Info("Helicopter slingload cargo Event Init.");
+
+private _vehicles = [];
+private _groups = [];
+private _difficult = random 10 < tierWar;
+
+private _player = selectRandom (call SCRT_fnc_misc_getRebelPlayers);
+
+if (isNil "_player") exitWith {
+    Error("No players found, aborting.");
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+};
+
+private _originPosition = position _player;
+
+Info_2("%1 will be used as center of the event at %2 position.", name _player, str _originPosition);
+
+private _potentialOutposts = (outposts + milbases + airportsX + resourcesX + factories) select {
+    sidesX getVariable [_x, sideUnknown] != teamPlayer && {(getMarkerPos _x) distance2D _player < distanceSPWN*5}
+};
+
+if (_potentialOutposts isEqualTo []) exitWith {
+    Info("No outposts in proximity, aborting Heli slingload cargo Event.");
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+};
+
+private _outpost = selectRandom _potentialOutposts;
+private _side = sidesX getVariable [_outpost, sideUnknown];
+private _faction = Faction(_side);
+private _outpostPosition = getMarkerPos _outpost;
+
+private _potentialspawnPosition = (outposts + milbases + airportsX + resourcesX + factories) select {
+    sidesX getVariable [_x, sideUnknown] != teamPlayer && sidesX getVariable [_x, sideUnknown] == _side && {(getMarkerPos _x) distance2D _player < distanceSPWN}
+};
+if (_potentialspawnPosition isEqualTo []) exitWith {
+    Info("No spawn positions in proximity, aborting Heli slingload cargo Event.");
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+};
+private _spawnPosition = selectRandom _potentialspawnPosition;
+if (_spawnPosition == _outpost) then {
+	while {true} do {
+		_potentialspawnPosition = (outposts + milbases + airportsX + resourcesX + factories) select {
+    	sidesX getVariable [_x, sideUnknown] != teamPlayer && sidesX getVariable [_x, sideUnknown] == _side  &&  {(getMarkerPos _x) distance2D _player < distanceSPWN}};
+	};
+	_spawnPosition = selectRandom _potentialspawnPosition;
+	if (_spawnPosition != _outpost) exitwith{};
+};
+
+private _actualspawnPosition = getMarkerPos _spawnPosition;
+
+private _HeliClass = selectRandom ((_faction get "vehiclesHelisLight") + (_faction get "vehiclesHelisTransport"));
+private _ammoBoxType = _faction get "ammobox";
+
+if (_HeliClass == "") exitWith {
+    Error("No Helicopters, problem with template, aborting Heli slingload cargo Event.");
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+};
+
+if (_HeliClass isKindOf "Heli_Transport_04_base_F") then {
+	_HeliClass = "O_Heli_Transport_04_F";
+};
```

Comment:
Why is this needed?

Reply:
CSAT helis that are in transport category, can't slingload because they already have pod, kinda, so I'm swapping them to actual transport heli

Reply:
Fair enough